Sources/PublicInterface/Client.swift

```diff
@@ -10,6 +10,12 @@ public protocol ClientDelegate: AnyObject {
     func client(_ client: Client, didConnect session: Session)
     func client(_ client: Client, didDisconnect session: Session)
     func client(_ client: Client, didUpdate session: Session)
+    func client(_ client: Client, willReconnect session: Session)
```

Comment:
do you want to move it to a separate Client2Delegate?

I don't remember the specifics (why I didn't go for the change to the ClientDelegate + default implementation), but the reason was to not break any source code that would update the library version.

I'll double check this anyway during review.

Reply:
It's a possibility too, I didn't have the latest changes when coding this new method. Although from a pure aesthetics point of view I kinda like it better in the original delegate protocol 😀 Up to you, really.

Reply:
Ok, I have checked out the branch in my main project, and the build fails because my existing class does not conform to protocol `ServerDelegate` now.

Please move this method to the `Client2Delegate` protocol, and the same for the ServerDelegate - to the `ServerDelegateV2`

Reply:
Ok, done!